tests/tensorflow/test_keras_model_export.py

```diff
@@ -693,18 +636,55 @@ def test_pyfunc_serve_and_score_transformers():
 def test_log_model_with_code_paths(model):
     artifact_path = "model"
     with mlflow.start_run(), mock.patch(
-        "mlflow.keras._add_code_from_conf_to_system_path"
+        "mlflow.tensorflow._add_code_from_conf_to_system_path"
     ) as add_mock:
-        mlflow.keras.log_model(model, artifact_path, code_paths=[__file__])
+        mlflow.tensorflow.log_model(model, artifact_path=artifact_path, code_paths=[__file__])
         model_uri = mlflow.get_artifact_uri(artifact_path)
-        _compare_logged_code_paths(__file__, model_uri, mlflow.keras.FLAVOR_NAME)
-        mlflow.keras.load_model(model_uri)
+        _compare_logged_code_paths(__file__, model_uri, mlflow.tensorflow.FLAVOR_NAME)
+        mlflow.tensorflow.load_model(model_uri)
         add_mock.assert_called()
 
 
 def test_virtualenv_subfield_points_to_correct_path(model, model_path):
-    mlflow.keras.save_model(model, path=model_path)
+    mlflow.tensorflow.save_model(model, path=model_path)
     pyfunc_conf = _get_flavor_configuration(model_path=model_path, flavor_name=pyfunc.FLAVOR_NAME)
     python_env_path = Path(model_path, pyfunc_conf[pyfunc.ENV]["virtualenv"])
     assert python_env_path.exists()
     assert python_env_path.is_file()
+
+
+def test_load_and_predict_keras_model_saved_by_mlflow128(tmpdir):
+    model_data_info, tracking_uri = save_or_log_keras_model_by_mlflow128(
+        str(tmpdir), task_type="log_model", save_as_type="keras"
+    )
+
+    model_uri = f"runs:/{model_data_info.run_id}/model"
+
+    def load_and_predict(load_model_fn):
+        with _use_tracking_uri(tracking_uri):
+            mlflow_model = load_model_fn()
+        predictions = mlflow_model.predict(model_data_info.inference_df)
+        np.testing.assert_allclose(predictions, model_data_info.expected_results_df)
+
+    load_and_predict(lambda: mlflow.pyfunc.load_model(model_uri))
+    load_and_predict(lambda: mlflow.tensorflow.load_model(model_uri))
+
+
+def test_load_tf_keras_model_with_options(tf_keras_model, model_path):
+    """
+    This is a backwards compatibility test to ensure that models saved in MLflow version <= 0.8.0
+    can be loaded successfully. These models are missing the `data` flavor configuration key.
+    """
```

Comment:
Is this comment correct?

Reply:
This test comes from existing code, I does not change it.

I think this line `del flavor_conf["data"]` means it mimic the <=0.8 version saved model. (These models are missing the `data` flavor configuration key.)

Reply:
I think this docstring comes from:

https://github.com/mlflow/mlflow/blob/95f2315b7d265382173396647324fdddd5901d7e/tests/keras/test_keras_model_export.py#L588-L605

Reply:
What this test does is to ensure a keras model is saved with `keras_model_kwargs`, right? Not to ensure a backwards compatibility with models saved in MLflow version <= 0.8.0.

Reply:
> What the current test does is ensure a keras model is saved with keras_model_kwargs

That's not the point in the test. This test we can remove keras_model_kwargs setting.
I think in the test the key part is here:

```
    flavor_conf = model_conf.flavors.get(mlflow.tensorflow.FLAVOR_NAME, None)
    assert flavor_conf is not None
    del flavor_conf["data"]
    model_conf.save(model_conf_path)

    model_loaded = mlflow.tensorflow.load_model(model_path)
```

It rewrite the saved MLmodel file and remove the "data" key from flavor conf (mimic mlflow<=0.8.0 saved format)
then reload the model.